drivers/gles2/rasterizer_scene_gles2.cpp

```diff
@@ -3285,6 +3285,30 @@ void RasterizerSceneGLES2::render_scene(const Transform &p_cam_transform, const
 			index++;
 		}
 
+		// for fog transmission, we want some kind of consistent ordering of lights
+		// add any more conditions here in which we need consistent light ordering
+		// (perhaps we always should have it, but don't know yet)
+		if (env && env->fog_transmit_enabled) {
+
+			struct _LightSort {
+
+				bool operator()(LightInstance *A, LightInstance *B) const {
+					return A > B;
```

Comment:
Will this result in a consistent sort across runs? I'm worried that each time the game is run this will result in a random sort order. If the user has a mix of dim and bright lights this would be very noticeable.

Reply:
Nope, as each time you run the memory addresses may well be different in relation to each other. But I suspect there is a similar problem with octree already - order of adding etc may be important. There may not be a perfect solution in that respect.

There may be something else we can use for more consistent sorting. We could just give lights an ever increasing ID (for each allocation) and sort on that? That might be better as it would be consistent within the run and between runs (providing everything was exactly the same). It's just storing an extra value with each light but I don't think that should matter.

Reply:
I've now added a per light counter as discussed above, a little bit more verbose, but should be consistent between runs of the game. :+1: